nano/node/json_handler.cpp

```diff
@@ -1214,20 +1213,19 @@ void nano::json_handler::block_confirm ()
 				nano::election_status status{ block_l, 0, 0, std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::system_clock::now ().time_since_epoch ()), std::chrono::duration_values<std::chrono::milliseconds>::zero (), 0, 1, 0, nano::election_status_type::active_confirmation_height };
 				node.active.recently_cemented.put (status);
 				// Trigger callback for confirmed block
-				auto account (node.ledger.account (transaction, hash));
-				bool error_or_pruned (false);
-				auto amount (node.ledger.amount_safe (transaction, hash, error_or_pruned));
+				auto account = node.ledger.account (*block_l);
```

Comment:
Not reusing existing db transaction - there are many call sites like this. Is there any reason for those changes? Reusing db transactions seems to me like a simple and positive thing, it's both faster and guarantees consistent view - something that might be important down the road with database sharding.

Reply:
These conversions from (tx, hash) to (block &) are misleading, they don't open new transactions on the database, these are utility functions that are hung on the ledger class.

I'm going to make a followup PR that moves these on to the block itself.

Reply:
Ahh, I see, it now makes sense. I should have started reviewing this PR by looking at the ledger class itself.